Sources/TensorFlow/Operators/NN.swift

```diff
@@ -142,46 +142,56 @@ func _vjpConv2D<Scalar: TensorFlowFloatingPoint>(
 ) -> (value: Tensor<Scalar>, pullback: (Tensor<Scalar>) -> (Tensor<Scalar>, Tensor<Scalar>)) {
     let value = conv2D(x, filter: filter, strides: strides, padding: padding, dilations: dilations)
     return (value, { v in
-        (conv2DBackpropInput(v, shape: x.shapeTensor, filter: filter,
-                             strides: strides, padding: padding, dilations: dilations),
+        (transposedConv2D(v, shape: x.shapeTensor, filter: filter,
+                          strides: strides, padding: padding, dilations: dilations),
          conv2DBackpropFilter(v, input: x, filterSizes: filter.shapeTensor,
                               strides: strides, padding: padding, dilations: dilations))
     })
 }
 
-/// TensorFlow builtin conv2d gradient helper for the input.
-@differentiable(wrt: (x, filter))
-@usableFromInline
-func conv2DBackpropInput<Scalar: TensorFlowFloatingPoint>(
-    _ x: Tensor<Scalar>,
+/// Returns a 2-D transposed convolution with the specified input, filter, strides, and padding.
```

Comment:
I'm not really for this renaming of gradient helper for the conv operation. It would be better if a separate wrapper for `transposedConv2D` is written which calls `conv2DBackpropInput`. This current renaming would also confuse people who check the source code about as to why transposed functions are being called for a convolution operation.

Reply:
I originally proposed to add alias for it in #596, and @dan-zheng proposed to just rename it.
Adding alias is less confusing for today's contributors, but it won't have much meaning for future contributors, so I decided to take rename way.

Reply:
The Tensorflow `Conv2DTranspose` calls the generated op as a wrapper, [here](https://github.com/tensorflow/tensorflow/blob/964a78f3834a17b44fb8898f3d3be95883fa0007/tensorflow/python/ops/nn_ops.py#L2218). Similarly done for `conv1dTranspose` and `conv3dTranpose`, While the conv2d [calls](https://github.com/tensorflow/tensorflow/blob/964a78f3834a17b44fb8898f3d3be95883fa0007/tensorflow/python/ops/nn_ops.py#L925), `input` and `filter` separately. 
A separation of concerns, with separate functions make sense and make for a cleaner api, than renaming `conv2dBackpropInput` as `transposedConv2D`.

Reply:
And there's separate `conv2d_backprop_input` [here](https://github.com/tensorflow/tensorflow/blob/964a78f3834a17b44fb8898f3d3be95883fa0007/tensorflow/python/ops/nn_ops.py#L2078).
`conv2d_transpose_v2` has extra operations (name_scope and data_format part), so they necessarily should be separated.
In S4TF there's no such operations. I'm not sure yet we should separate them.

Reply:
The `name_scope` and `data_format` isn't exactly an operation as they are related to input type. We currently only support the NHWC format, I think once we are thoroughly done implementing layers, We should start looking to support the NCWH format.